src/hooks/UseMutation.re

```diff
@@ -13,130 +13,90 @@ type executeMutation =
     ~meta: Types.operationDebugMeta=?,
     ~suspense: bool=?,
     ~preferGetMethod: bool=?,
-    unit
+    'variables
   ) =>
-  Js.Promise.t(Types.operationResult(Js.Json.t));
+  Js.Promise.t(Types.operationResult('data));
 
-type useMutationResponseJs = (
-  Types.hookResponseJs(Js.Json.t),
-  executeMutationJs,
+type useMutationResponseJs('jsData) = (
+  Types.hookResponseJs('jsData),
+  executeMutationJs('jsData),
 );
 
-type useMutationResponse('response) = (
-  Types.hookResponse('response),
-  executeMutation,
+type useMutationResponse('variables, 'data) = (
+  Types.hookResponse('data),
+  executeMutation('variables, 'data),
 );
 
 [@bs.module "urql"]
-external useMutationJs: string => useMutationResponseJs = "useMutation";
+external useMutationJs: string => useMutationResponseJs('jsData) =
+  "useMutation";
 
 /**
  * The useMutation hook.
- *
- * Accepts the following arguments:
- *
- * request – a Js.t containing the query and variables corresponding
- * to the GraphQL mutation, and a parse function for decoding the JSON response.
  */
-let useMutation = (~request) => {
-  let query = request##query;
-  let variables = request##variables;
-  let parse = request##parse;
+let useMutation:
+  type data variables.
+    (
+      ~query: (module Types.Operation with
```

Comment:
I'd advocate for one of two options here:

1. Rename this labelled argument to `mutation`.

```suggestion
      ~mutation: (module Types.Operation with
```

2. Don't bother with a labelled argument, since the hook only accepts a single argument.

1 is more compatible with the current API, while 2 maybe makes more sense from a language perspective. What do you think?

Reply:
Both compiles to same js.
I actually like the first approach for consistency between APIs.
AFAIK there shoudn't be any problem with unary function with a (explicitly typed) labeled argument.
you can even omit label (though compiler gives a warning):
```reason
let (state, exec) = Hooks.useMutation((module MyMutation));
```

maybe @jfrolich can also help with this.

Reply:
I don't think it makes a difference. If you have multiple arguments it's good to have the query as the first argument because of type inference I think. Otherwise the variable record might not be inferred (not 100% sure).